pkg/controller/core/clusterqueue_controller.go

```diff
@@ -160,6 +160,9 @@ func (h *assignedWorkloadHandler) Update(e event.UpdateEvent, q workqueue.RateLi
 	if newW.Spec.Admission != nil && (oldW.Spec.Admission == nil || newW.Spec.Admission.ClusterQueue != oldW.Spec.Admission.ClusterQueue) {
 		q.AddAfter(requestForWorkloadClusterQueue(newW), constants.UpdatesBatchPeriod)
 	}
+	if newW.Spec.Admission != nil && workloadFinished(newW) {
```

Comment:
@alculquicondor  😃

Reply:
did this reduce the test flakiness?

Reply:
Yes sir :)

Reply:
Ah, good catch! Better put it in line 160 though (with an OR). Otherwise you might be queueing twice.

Reply:
done :)